pkg/plugincmdtree/cache.go

```diff
@@ -13,13 +17,12 @@ import "github.com/vmware-tanzu/tanzu-cli/pkg/cli"
 type Cache interface {
 	// GetTree returns the plugin command tree
 	// If the plugin command tree doesn't exist, it constructs and adds the command tree to the cache
-	// and then returns the plugin command tree, otherwise it returns error
-	GetTree(plugin *cli.PluginInfo) (*CommandNode, error)
-	// ConstructAndAddTree constructs and adds the plugin command tree to the cache
-	// If the plugin command tree already exists, it returns success immediately
-	ConstructAndAddTree(plugin *cli.PluginInfo) error
-	// DeleteTree deletes the plugin command tree from the cache
-	DeleteTree(plugin *cli.PluginInfo) error
+	// and then returns the plugin command tree, otherwise it returns an error
+	GetTree(rootCmd *cobra.Command, plugin *cli.PluginInfo) (*CommandNode, error)
+	// DeletePluginTree deletes the plugin command tree from the cache
+	DeletePluginTree(plugin *cli.PluginInfo) error
```

Comment:
Nit: Function name `DeletePluginTree` is not opposite to `GetTree`. `GetTree` get the plugin command tree( not all plugin command trees in cache) where as `DeleteTree` deletes all the plugin command trees from cache.

Reply:
Good point.  Ok for me to rename `GetTree` to `GetPluginTree`?